utils/s2n_blob.h

```diff
@@ -30,11 +30,16 @@ struct s2n_blob {
     /* The amount of memory allocated for this blob (i.e. the amount of memory
      * which needs to be freed when the blob is cleaned up). If this blob was
      * created with s2n_blob_init(), this value is 0. If s2n_alloc() was called,
-     * this value will be greater than 0.
+     * this value will be greater than or equal to size.
+     * 
+     * size < allocated implies that an allocated blob is being reused to store 
+     * a smalled amount of data.
      */
     uint32_t allocated;
 
-    /* Can this blob be resized */
+    /* An allocated blob (e.g.`s2n_alloc`) is always growable. A "reference"
+     * blob (from `s2n_init`) is never growable.
+     */
```

Comment:
I'm not sure whether this behavior was actually intentional :thinking: Kind of seems like it was an artifact of aliasing "s2n_alloc" to "s2n_realloc", probably for unrelated reasons.

Separating "growable" from "allocated" definitely makes sense in the stuffer, because "allocated" indicates whether or not the data needs to be freed, and "growable" indicates whether insufficient space should be an error or lead to an allocation (usually it's an error).

But I guess with a blob, it wouldn't grow automatically, and you'd have to explicitly call "realloc", which is pretty strong indicator that you want to allocate more memory :)

Reply:
Ya, I think "growable" is just a very confusing concept for the blob, and don't understand a scenario where
1. a stuffer would have `growable=false`
2. a blob would have `growable=true`
or vice versa. 

I opened #4944 to track further "normalization".